src/test/java/TasksServletTest.java

```diff
@@ -0,0 +1,146 @@
+  private StringWriter stringWriter;
+  private PrintWriter printWriter;
+
+  // Tasks must be returned in order of retrieval - JSON includes tasks in desired order
+  private static final String TASK_TITLE_ONE = "task one";
+  private static final String TASK_TITLE_TWO = "task two";
+  private static final String TASK_TITLE_THREE = "task three";
+  private static final String TASK_TITLE_FOUR = "task four";
+  private static final String TASK_ONE_TWO_JSON =
+      String.format("[{\"title\":\"%s\"},{\"title\":\"%s\"}]", TASK_TITLE_ONE, TASK_TITLE_TWO);
+  private static final String TASK_ALL_JSON =
+      String.format(
+          "[{\"title\":\"%s\"},{\"title\":\"%s\"},{\"title\":\"%s\"},{\"title\":\"%s\"}]",
+          TASK_TITLE_ONE, TASK_TITLE_TWO, TASK_TITLE_THREE, TASK_TITLE_FOUR);
+  private static final String EMPTY_JSON = "[]";
+  private static final List<TaskList> NO_TASK_LISTS = ImmutableList.of();
+  private static final List<TaskList> SOME_TASK_LISTS =
+      ImmutableList.of(new TaskList(), new TaskList());
+  private static final List<Task> NO_TASKS = ImmutableList.of();
+  private static final List<Task> TASKS_ONE_TWO =
+      ImmutableList.of(new Task().setTitle(TASK_TITLE_ONE), new Task().setTitle(TASK_TITLE_TWO));
+  private static final List<Task> TASKS_THREE_FOUR =
+      ImmutableList.of(new Task().setTitle(TASK_TITLE_THREE), new Task().setTitle(TASK_TITLE_FOUR));
+
+  @Before
+  public void setUp() throws IOException, GeneralSecurityException {
+    authenticationVerifier = Mockito.mock(AuthenticationVerifier.class);
+    tasksClientFactory = Mockito.mock(TasksClientFactory.class);
+    tasksClient = Mockito.mock(TasksClient.class);
+    servlet = new TasksServlet(authenticationVerifier, tasksClientFactory);
+
+    Mockito.when(tasksClientFactory.getTasksClient(Mockito.any())).thenReturn(tasksClient);
+    // Authentication will always pass
+    Mockito.when(authenticationVerifier.verifyUserToken(Mockito.anyString()))
+        .thenReturn(AUTHENTICATION_VERIFIED);
+
+    request = Mockito.mock(HttpServletRequest.class);
+    response = Mockito.mock(HttpServletResponse.class);
+    Mockito.when(request.getCookies()).thenReturn(validCookies);
+
+    // Writer used in get/post requests to capture HTTP response values
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+    Mockito.when(response.getWriter()).thenReturn(printWriter);
+  }
+
+  @Test
+  public void noTaskLists() throws IOException, ServletException {
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(NO_TASK_LISTS);
+    servlet.doGet(request, response);
+    printWriter.flush();
+    Assert.assertTrue(stringWriter.toString().contains(EMPTY_JSON));
+  }
+
+  @Test
+  public void emptyTaskLists() throws IOException, ServletException {
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(SOME_TASK_LISTS);
+    Mockito.when(tasksClient.listTasks(Mockito.any())).thenReturn(NO_TASKS);
```

Comment:
Can you verify that `listTasks` was passed the correct task list object? Same for the other tests below...

Reply:
Good call - done